### PR TITLE
RR-584 - Skeleton cypress tests for new Goals summary card scenarios on W&S tab

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR does a little refactoring to the cypress wiremock stubs in respect of the Curious API (to make it easier for a cypress test to say it wants one that returns goals, one that returns no goals, or one that returns an error); and then adds the skeleton methods for new cypress tests that we are about to implement
The new cypress tests are all defined with `it.skip` for the time being - they have the correct scenario description and setup in respect of wiremock stubs; the last thing they need is for the relevant assertions to be written for the rendered page content which will come after @chrimesdev 's current work

### CODE REVIEW

File: integration_tests/e2e/vc2GoalsPage.cy.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ context('Prisoners VC2 goals page', () => {
 
   it('should contain a banner about learning and work progress service replacing VC2', () => {
     // Given
-    cy.setupWorkAndSkillsPageStubs({ prisonerNumber: 'G6123VU', emptyStates: false })
+    cy.setupWorkAndSkillsPageStubs({ prisonerNumber: 'G6123VU' })
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/vc2GoalsPage.cy.ts:17]: I've set `emptyStates: false` as the default (further down), so no need for calls to this stub to pass it (unless a given test wants to change the behaviour to get an "empty state")

File: integration_tests/e2e/workAndSkillsPage/workAndSkillsGoals.cy.ts [hunk 2]
```diff
@@ -103,4 +106,60 @@ context('Work and skills page - Goals card', () => {
     workAndSkillsPage.GoalsLongTermText().should('exist')
     workAndSkillsPage.GoalsLongTermText().contains('The prisoner does not have any long term goals.')
   })
+
+  it.skip('should display the Goals card with populated goals given prisoner has PLP goals only', () => {
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/workAndSkillsPage/workAndSkillsGoals.cy.ts:110]: These are the new scenarios, all ready and waiting for their assertions to be fleshed out 😁

File: integration_tests/mockApis/curiousApi.ts [hunk 3]
```diff
@@ -73,18 +73,9 @@ export default {
...
-        ? aValidLearnerGoals({ prn: prisonerNumber })
-        : aValidLearnerGoals({
-            prn: prisonerNumber,
-            employmentGoals: [],
-            personalGoals: [],
-            longTermGoals: [],
-            shortTermGoals: [],
-          })
+
+  stubGetCuriousGoals: (prisonerNumber = 'G6123VU') => {
```
Comment (nathanrussell-moj-digital) [inline integration_tests/mockApis/curiousApi.ts:77]: I've changed this `stubGetCuriousGoals` stub so that it returns data that represents a prisoner's goals
I have added separate stubs below for a prisoner with no goals, and a 500 error
That way the cypress test itself has to be explicit about which stub it wants to use, which increases the readability of the test 👍

File: integration_tests/support/commands.ts [hunk 4]
```diff
@@ -74,12 +74,12 @@ Cypress.Commands.add('setupAlertsPageStubs', ({ bookingId, prisonerNumber, priso
...
 })
 
-Cypress.Commands.add('setupWorkAndSkillsPageStubs', ({ prisonerNumber, emptyStates }) => {
+Cypress.Commands.add('setupWorkAndSkillsPageStubs', ({ prisonerNumber, emptyStates = false }) => {
   cy.task('stubGetLearnerEmployabilitySkills', prisonerNumber)
   cy.task('stubGetLearnerEducation', prisonerNumber)
   cy.task('stubGetLearnerProfile', prisonerNumber)
   cy.task('stubGetLearnerLatestAssessments', prisonerNumber)
-  cy.task('stubGetCuriousGoals', { prisonerNumber, emptyStates })
+  cy.task('stubGetCuriousGoals', prisonerNumber)
```
Comment (nathanrussell-moj-digital) [inline integration_tests/support/commands.ts:82]: I've removed the ability to pass "empty states" into this stub (see explanation above)

File: integration_tests/support/commands.ts [hunk 5]
```diff
@@ -74,12 +74,12 @@ Cypress.Commands.add('setupAlertsPageStubs', ({ bookingId, prisonerNumber, priso
...
   cy.task('stubGetLearnerNeurodivergence', prisonerNumber)
   cy.task('stubGetOffenderAttendanceHistory', prisonerNumber)
   cy.task('stubGetOffenderActivities', { prisonerNumber, emptyStates })
```
Comment (nathanrussell-moj-digital) [inline integration_tests/support/commands.ts:85]: I've left this stub alone; so in respect of the work and skills page stubs, it is the only one now where you can control "empty states" like this
(we are doing Goals, not Offender Activities, so I didnt want to mess with it)

**Review by mattwills-moj-digital — APPROVED** [review]
LGTM

**Review by leejacobson-moj — APPROVED** [review]
Looks good to me. Approved.
